hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/fs/CachingSpaceUsageSource.java

```diff
@@ -137,9 +139,24 @@ private void refresh() {
     //only one `refresh` can be running at a certain moment
     if (isRefreshRunning.compareAndSet(false, true)) {
       try {
-        cachedValue.set(source.getUsedSpace());
+        long newUsedSpace = source.getUsedSpace();
+        // Check for negative value before setting it to the cache
+        if (newUsedSpace >= 0) {
+          cachedValue.set(newUsedSpace);
+        } else {
+          LOG.error(
+              "Received negative used space value: {}. Keeping the last known good value: {}.",
+              newUsedSpace, cachedValue.get());
+        }
+      } catch (UncheckedIOException e) {
+        // Log the error and do not update the cached value
+        LOG.error(
+            "Error refreshing space usage for {}. Keeping the last known good value: {}",
+            source, cachedValue.get(), e);
```

Comment:
There is no need to duplicate the `catch` block, since `UncheckedIOException` is a `RuntimeException`.  (Also, if it weren't, a single clause could `catch` multiple exceptions.)

Reply:
I have removed it thanks!